Eastern-Research-Group/mywaterway
# Feature/further update streamgages

## Related Issues:
* https://jira.epa.gov/browse/HMW-4

## Main Changes:
Further refines display of USGS streamgages:
* Organizes streamgages returned from STA web service into primary category ("top ten" initially displayed), secondary category (displayed below primary after a show/hide button), and exclude list (never displayed)
* Links parameters in the "primary" category to their terms found in the glossary
* Updates USGS STA parameter mapping file to include more HMW specific names, units, and updated ordering for parameters
* Converts a few parameters that return values in Celsius to Fahrenheit
* Replaced multiple measurements of a single parameter returned in STA web service with a message
* Only displays gage height number on map if a single gage height measurement is returned
* Renames streamgages labels to "Current Water Conditions"
* Adds intro text to Monitoring & Sensors subtab
* Only shows the "Monitoring Measurements" sort option if monitoring locations switch is checked
* Removes sub tabs from Community Monitoring page
* Moves USGS streamgages content out of a function returning JSX in WaterbodyInfo component into its own component

## Steps To Test:
For all of the links below, switch to the "Monitoring & Sensors" subtab, and toggle on the "Current Water Conditions" switch (you don't need to, but I like to turn off the waterbodies and surrounding waterbodies layers too, as its easier to focus on just the streamgages then):
1. Navigate to http://localhost:3000/community/22201/overview
2. Verify that the "Water level" parameter indicates "multiple measurements" instead of showing two water level measurements (NOTE: the version currently on the Cloud.gov dev site would only include water levels that had a specific WebDescription property, but we chose this implementation as it's more universal because it: 1) works for all parameters...not just water level, and 2) some gages have water level data returned, but they don't all have the specific WebDescription property value we were previously filtering on)
3. Navigate to http://localhost:3000/community/030801031304/overview
4. Verify two primary parameters are shown, followed by a show/hide button that toggles any secondary parameters (in this case one).
5. Also verify the °F water temperature measurement displayed in HMW equals the °C water temperature shown on the USGS page ("More information" link)
6. Navigate to http://localhost:3000/community/020700100102/overview
7. Verify terms displayed link to the glossary (this is true for other links as well, this location just have several parameters in the "primary" category)

## TODO:
- [ ] Access USGS Daily Values Web Service to fetch precipitation data if it exists for each streamgage, and display it along with the other "parameters" in the primary/initial category
- [ ] Discuss edge cases with parameters (none returned in STA web service and no primary params returned in STA web service) with EPA
- [ ] Update `WaterbodyInfo` to not directly include JSX or functions returning JSX for each "type" passed – move that rendered content into their own components (kept in same file), as is done with `<UsgsStreamgagesContent />` here
